test/suites/clustering.sh

```diff
@@ -3951,6 +3951,97 @@ test_clustering_evacuation() {
   LXD_NETNS=
 }
 
+test_clustering_evacuation_restore_operations() {
+  local LXD_DIR
+
+  echo "Create cluster with 2 nodes"
+  setup_clustering_bridge
+  prefix="lxd$$"
+  bridge="${prefix}"
+
+  # The random storage backend is not supported in clustering tests,
+  # since we need to have the same storage driver on all nodes, so use the driver chosen for the standalone pool.
+  poolDriver=$(lxc storage show "$(lxc profile device get default root pool)" | awk '/^driver:/ {print $2}')
+
+  # Spawn first node
+  setup_clustering_netns 1
+  LXD_ONE_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  ns1="${prefix}1"
+  spawn_lxd_and_bootstrap_cluster "${ns1}" "${bridge}" "${LXD_ONE_DIR}" "${poolDriver}"
+
+  # Add a newline at the end of each line. YAML has weird rules.
+  cert=$(sed ':a;N;$!ba;s/\n/\n\n/g' "${LXD_ONE_DIR}/cluster.crt")
+
+  # Spawn a second node
+  setup_clustering_netns 2
+  LXD_TWO_DIR=$(mktemp -d -p "${TEST_DIR}" XXX)
+  ns2="${prefix}2"
+  spawn_lxd_and_join_cluster "${ns2}" "${bridge}" "${cert}" 2 1 "${LXD_TWO_DIR}" "${LXD_ONE_DIR}" "${poolDriver}"
+
+  LXD_DIR="${LXD_ONE_DIR}"
+  ensure_import_testimage
+
+  echo "Launch 3 containers on node1"
+  for i in c{1..3}; do lxc launch testimage "${i}" --target node1; done
+
+  echo "Start node1 evacuation in background"
+  lxc cluster evacuate node1 --quiet --force &
+  sleep 1 # Wait a bit for the operation to start
+
+  echo "Check restore fails while evacuation operation in progress"
+  ! lxc cluster restore node1 --quiet --force || false
+
+  echo "Wait for all containers to be evacuated"
+  for _ in {1..60}; do
+    all_evacuated=true
+    for c in c1 c2 c3; do
+      loc=$(lxc list -f csv -c L "${c}")
+      if [ "${loc}" = "node1" ]; then
+        all_evacuated=false
+        break
+      fi
+    done
+    $all_evacuated && break
+    sleep 1
+  done
+  sleep 1 # Wait a bit for the operation to complete
+
+  echo "Verify all containers are no longer on node1"
+  for c in c1 c2 c3; do
+    [ "$(lxc list -f csv -c L "${c}")" != "node1" ]
+  done
+
+  echo "Start node1 restore in background"
+  lxc cluster restore node1 --quiet --force &
+  sleep 1 # Wait a bit for the operation to start
+
+  echo "Check evacuation fails while restore operation in progress"
+  ! lxc cluster evacuate node1 --quiet --force || false
```

Comment:
After looking at the Go code, I think that checking the specific error message is returned would be ideal here. Safer than just expecting a failure.